.github/workflows/dotnetcore.yml

```diff
@@ -1,32 +1,63 @@
 name: .NET Core
 
-on: [ push ]
+on:
+  push:
+    branches:
+      - main
+      - sdk-automation/models
+      - promote/main
+  pull_request:
+    branches:
+      - main
+      - sdk-automation/models
+      - promote/main
+  workflow_dispatch: {}
 
 jobs:
-  build:
-    name: Build and test
-
+  build-test:
+    name: Build and Test
     runs-on: ${{ matrix.os }}
+    
     strategy:
       matrix:
         os: [ ubuntu-latest, windows-latest ]
+        dotnet-version: [ '6.0.x' ]
+
     steps:
       - uses: actions/checkout@v4
+      
       - name: Setup .NET
         uses: actions/setup-dotnet@v4
         with:
-          dotnet-version: |
-            6.0.x
-      - name: Install tools
-        run: |
-          dotnet restore
-
+          dotnet-version: ${{ matrix.dotnet-version }}
+        
       - name: Build Debug
         run: dotnet build -c Debug
-
-      - name: Run test suite net6.0
+        
+      - name: Run unit tests
         run: dotnet test --no-build -c Debug -f net6.0 Adyen.Test/Adyen.Test.csproj
 
-      - name: Run integration test suite net6.0
-        run: dotnet test --no-build -c Debug -f net6.0 Adyen.IntegrationTest/Adyen.IntegrationTest.csproj
-        if: ${{ contains(github.event.pull_request.labels.*.name, 'release') }}
\ No newline at end of file
+  integration-tests:
+    name: Integration Tests
+    runs-on: ubuntu-latest
+    needs: build-test
+    
+    #Check if event is a pull resquest and has a release tag and is not from a forked repo
```

Comment:
nit: `# comment`-style (add a space after `#`)
& typo in `pull request`

Reply:
Good catch. Thanks :)